src/Validator/ValidBasePackageVersionValidator.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the package t3o/get.typo3.org.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ *
+ * The TYPO3 project - inspiring people to share!
+ */
+
+namespace App\Validator;
+
+use App\Entity\Sitepackage;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+class ValidBasePackageVersionValidator extends ConstraintValidator
+{
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$value instanceof Sitepackage) {
+            throw new UnexpectedValueException($value, Sitepackage::class);
+        }
+
+        if (!$constraint instanceof ValidBasePackageVersion) {
+            throw new UnexpectedValueException($constraint, ValidBasePackageVersion::class);
+        }
+
+        if ($value->getBasePackage() === 'site_package_tutorial' && $value->getTypo3Version() !== 13.4) {
```

Comment:
i think we need to check here for versions available in the packages directory and have it generally available as a validator. this one currently only checks for `site_package_tutorial` and has a fixed version constraint

Reply:
will check the validator does not work with the api endpoint currently - will also check how to add general support.


Payload: 

```
{
  "base_package": "site_package_tutorial",
  "typo3_version": 12.4,
  "vendor_name": "string",
  "vendor_name_alternative": "string",
  "title": "My Sitepackage",
  "description": "Project Configuration for Client",
  "package_name": "string",
  "package_name_alternative": "string",
  "extension_key": "string",
  "repository_url": "https://github.com/FriendsOfTYPO3/introduction",
  "author": {
    "name": "J. Doe",
    "email": "info@typo3.com",
    "company": "TYPO3",
    "homepage": "https://typo3.com"
  }
}
```

Reply:
The api end point does not seem to validate whatoever so this seems to be a different issue: 
```
curl -X 'POST' \
  'https://get.typo3.org/api/v1/sitepackage/' \
  -H 'accept: application/zip' \
  -H 'Content-Type: application/json' \
  -d '{
  "base_package": "bootstrap_package",
  "typo3_version": 8.7,
  "vendor_name": "string",
  "vendor_name_alternative": "string",
  "title": "My Sitepackage",
  "description": "Project Configuration for Client",
  "package_name": "string",
  "package_name_alternative": "string",
  "extension_key": "string",
  "repository_url": "https://github.com/FriendsOfTYPO3/introduction",
  "author": {
    "name": "J. Doe",
    "email": "info@typo3.com",
    "company": "TYPO3",
    "homepage": "https://typo3.com"
  }
}'
``` 
just gives you a 500 currently